services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/claimprovider/GA4GHPassportClaimProvider.java

```diff
@@ -67,6 +79,24 @@ public class GA4GHPassportClaimProvider implements OIDCClaimProvider {
 	@Autowired
 	private Clock clock;
 	
+	@Autowired
+	private StackConfiguration stackConfiguration;
+	
+	private String oidcSignatureKeyId;
+	private PrivateKey oidcSignaturePrivateKey;
+	
+	public static final String VISA_CLAIM_NAME = "ga4gh_visa_v1";
+	
+	@Override
+	public void afterPropertiesSet() {
+		List<String> pemEncodedRsaPrivateKeys = stackConfiguration.getOIDCSignatureRSAPrivateKeys();
+		
+		// grab the latest private key to be used for signing
+		KeyPair keyPair = KeyPairUtil.getRSAKeyPairFromPrivateKey(pemEncodedRsaPrivateKeys.get(pemEncodedRsaPrivateKeys.size()-1));
+		this.oidcSignaturePrivateKey=keyPair.getPrivate();
```

Comment:
In general we do not want to keep secrets in memory for a brief amount of time (You can see an example in the stack configuration)